solvcon/block.py

```diff
@@ -292,6 +292,17 @@ def __init__(self, *args, **kw):
         # sanity check.
         self.check_sanity()
 
+    def __getstate__(self):
+        return self.__dict__
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        for name in self.TABLE_NAMES:
```

Comment:
Both the array and the table were unpickled.  Here we abandon the array versions.